src/lib.rs

```diff
@@ -204,6 +203,110 @@ mod zerocopy {
     pub(crate) use crate::*;
 }
 
+/// The layout of a type which might be dynamically-sized.
+///
+/// `DstLayout` describes the layout of sized types, slice types, and "custom
+/// DSTs" - ie, those that are known by the type system to have a trailing slice
+/// (as distinguished from `dyn Trait` types - such types *might* have a
+/// trailing slice type, but the type system isn't aware of it).
+#[doc(hidden)]
+#[allow(missing_debug_implementations, missing_copy_implementations)]
+#[cfg_attr(test, derive(Debug, PartialEq, Eq))]
+pub struct DstLayout {
+    /// For sized types, `size_of::<T>()`. For DSTs, the size of the type when
+    /// the trailing slice field contains 0 elements.
+    _fixed_prefix_size: usize,
+    /// The alignment of the type.
+    _align: NonZeroUsize,
```

Comment:
These fields could be collapsed into a single `core::alloc::Layout` field:
```suggestion
    /// For sized types, `Layout::new::<T>()`. For DSTs, the it's the layout of
    /// the type when the trailing slice field contains 0 elements.
    _base_layout: core::alloc::Layout,
```
Doing so would simplify the definition of `DstLayout::for_type`, and give us a space-optimized representation of alignment.

Reply:
Great idea, done!